ts/WoltLabSuite/WebComponent/woltlab-core-pagination.ts

```diff
@@ -0,0 +1,318 @@
+      }
+      ul.append(this.#getLinkItem(this.count));
+
+      const nextLinkElement = this.#getNextLinkElement();
+      if (nextLinkElement) {
+        nav.append(nextLinkElement);
+      }
+    }
+
+    #getNavElement(): HTMLElement {
+      const nav = document.createElement("nav");
+      nav.setAttribute("role", "navigation");
+      nav.ariaLabel = window.WoltLabLanguage.getPhrase("wcf.page.pagination");
+      nav.classList.add(this.#className);
+
+      return nav;
+    }
+
+    #getPreviousLinkElement(): HTMLDivElement | undefined {
+      if (this.page === 1) {
+        return;
+      }
+
+      const div = document.createElement("div");
+      div.classList.add(this.#className + "__prev");
+
+      const button = this.#getButtonElement(this.page - 1);
+      if (button instanceof HTMLAnchorElement) {
+        button.rel = "prev";
+      }
+      button.title = window.WoltLabLanguage.getPhrase("wcf.global.page.previous");
+      button.classList.add("jsTooltip");
+      div.append(button);
+
+      const icon = document.createElement("fa-icon");
+      icon.setIcon("arrow-left");
+      icon.ariaHidden = "true";
+      button.append(icon);
+
+      return div;
+    }
+
+    #getNextLinkElement(): HTMLDivElement | undefined {
+      if (this.page === this.count) {
+        return;
+      }
+
+      const div = document.createElement("div");
+      div.classList.add(this.#className + "__next");
+
+      const button = this.#getButtonElement(this.page + 1);
+      if (button instanceof HTMLAnchorElement) {
+        button.rel = "next";
+      }
+      button.title = window.WoltLabLanguage.getPhrase("wcf.global.page.next");
+      button.classList.add("jsTooltip");
+      div.append(button);
+
+      const icon = document.createElement("fa-icon");
+      icon.setIcon("arrow-right");
+      icon.ariaHidden = "true";
+      button.append(icon);
+
+      return div;
+    }
+
+    #getButtonElement(page: number): HTMLAnchorElement | HTMLButtonElement {
+      let button: HTMLAnchorElement | HTMLButtonElement;
+      const url = this.getLinkUrl(page);
+
+      if (url) {
+        button = document.createElement("a");
+        button.href = url;
+      } else {
+        button = document.createElement("button");
+        button.type = "button";
+
+        if (this.page === page) {
+          button.disabled = true;
+        } else {
+          button.addEventListener("click", () => {
+            this.#switchPage(page);
+          });
+        }
+      }
+
+      button.classList.add(this.#className + "__link");
+
+      return button;
+    }
+
+    #getLinkItem(page: number): HTMLLIElement {
+      const li = document.createElement("li");
+      li.classList.add(this.#className + "__item");
+
+      const button = this.#getButtonElement(page);
+      button.ariaLabel = window.WoltLabLanguage.getPhrase("wcf.page.pageNo", { pageNo: page });
+      if (page === this.page) {
+        button.ariaCurrent = "page";
```

Comment:
The `ariaCurrent` property is not supported by Firefox: https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaCurrent. This will cause issues, because the CSS uses the `aria-current` attribute, instead of the `pagination__link--current` class.